### PR TITLE
Allow an existing secret to be used

### PR DESCRIPTION
**Pull request by darend**
## what

Allow an existing secret to be used instead of always creating one

## why

Some workflows prefer managing secrets separately from the application. For example using sops secrets 

## testing

- Installed chart without this change and then performed a diff with this change applied. The diff was clean except for added quotes around references to secrets
- Set `yourlsPasswordExistingSecret` and `externalDatabase.existingSecret`. The two secrets were deleted, and the references were updated to use the provided values

### CODE REVIEW

File: charts/yourls/templates/_helpers.tpl [hunk 1]
```diff
@@ -4,7 +4,7 @@ Expand the name of the chart.
 */}}
 {{- define "yourls.name" -}}
 {{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
-{{- end -}}
+ {{- end -}}
```
Comment (dgw) [inline charts/yourls/templates/_helpers.tpl:7]: Accidental change?

Comment (darend) [reply charts/yourls/templates/_helpers.tpl]: https://github.com/YOURLS/charts/pull/22/commits/59d604b8aef5af9ebca1635798e28c9a5a6c5e36

**Comment by dgw** [comment]
I'll let @LeoColomb have final say, but my very limited knowledge of Helm says this looks good.

**Comment by LeoColomb** [comment]
Thanks a lot for your PR @darend! (and sorry for my quite late answer)
Looks good to me, but just want to be sure variable naming keeps following standard, do you have some other chart example when these kind of variables are used?

**Comment by darend** [comment]
> Looks good to me, but just want to be sure variable naming keeps following standard, do you have some other chart example when these kind of variables are used?

The most common pattern is `existingSecret`

- Datadog uses `xyzExistingSecret` https://github.com/DataDog/helm-charts/tree/master/charts/datadog#create-and-provide-a-secret-that-contains-your-datadog-api-key
- Bitnami charts use `existingSecret`, example from Redis https://github.com/bitnami/charts/blob/master/bitnami/redis/values.yaml#L211
- Jenkins uses `existingSecret ` https://github.com/jenkinsci/helm-charts/blob/main/charts/jenkins/values.yaml#L730
- Grafana uses `existingSecret` https://github.com/grafana/helm-charts/blob/main/charts/grafana/values.yaml#L278
- Prometheus uses `configFromSecret` https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml#L98

**Review by LeoColomb — APPROVED** [review]

**Comment by LeoColomb** [comment]
Perfect then 😄 
Thanks again!

**Pull request merged**